packages/go_router/lib/src/router.dart

```diff
@@ -188,6 +183,13 @@ class GoRouter extends ChangeNotifier implements RouterConfig<RouteMatchList> {
         queryParams: queryParams,
       );
 
+  /// Get the location for the provided route.
+  ///
+  /// Builds the absolute path for the route, by concatenating the paths of the
+  /// route and all its ancestors.
+  String? locationForRoute(RouteBase route) =>
```

Comment:
this can be private?

Reply:
I don't see how, as it is used in StackedNavigationShellState. I added this to avoid having to drill down deeper into the dependencies of GoRouter (i.e. `GoRouter.routeInformationParser.configuration.locationForRoute`).